### PR TITLE
PR for https://github.com/kr1sp1n/node-vault/issues/60

### PR DESCRIPTION
**Pull request by firefoxNX**
PR for https://github.com/kr1sp1n/node-vault/issues/60
use native promise instead of bluebird
removed babel use node 8
use request-promise-native

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,4 +16,4 @@ script:
   - npm run coverage
 
 after_success:
-  - codecov
+  - codecov
```
Comment (kr1sp1n) [inline .travis.yml]: Please add a newline at end of file.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM mhart/alpine-node:6
+FROM node:8
```
Comment (kr1sp1n) [inline Dockerfile]: Please use `node:8.6-alpine` for smaller images:
`node:8`= 269 MB
`node:8.6-alpine` = 23 MB
See https://hub.docker.com/r/library/node/tags/

File: .travis.yml (hunk 1 above)
Comment (firefoxNX) [reply .travis.yml]: done

File: Dockerfile (hunk 2 above)
Comment (firefoxNX) [reply Dockerfile]: done

File: package.json [hunk 3]
```diff
@@ -11,29 +11,24 @@
     "test-watch": "mocha --watch",
     "predocs": "node scripts/write-features.js",
     "docs": "docco --output docs src/*.js",
-    "build": "babel --presets es2015 --out-dir dist src",
```
Comment (kr1sp1n) [inline package.json]: That means that babel and related presets are not longer needed in the `devDependencies`, right?

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,9 @@
 sudo: required
 
+language: node_js
+node_js:
+  - "8"
```
Comment (kr1sp1n) [inline .travis.yml]: What about backward compatibility to node 6 and 7? Without babel compiler other apps running below node v8 are not working, right?

File: package.json (hunk 3 above)
Comment (firefoxNX) [reply package.json]: Yes. I removed babel from devDependencies.

Comment (kr1sp1n) [reply package.json]: 👍

File: .travis.yml (hunk 4 above)
Comment (firefoxNX) [reply .travis.yml]: Yes although native Promise support has existed since node 4 - http://node.green/#ES2015-built-ins-Promise

Comment (firefoxNX) [reply .travis.yml]: I test with ```FROM node:6.11-alpine``` locally and the tests passed.

Comment (kr1sp1n) [reply .travis.yml]: Me too 😉

**Pull request merged**